frontend/app/src/util/AppNavigation.test.ts

```diff
@@ -0,0 +1,307 @@
+    "@streamlit/app/src/SegmentMetricsManager"
+  )
+
+  const MockedClass = jest.fn().mockImplementation((...props) => {
+    const metricsMgr = new actualModule.SegmentMetricsManager(...props)
+    jest.spyOn(metricsMgr, "enqueue")
+
+    return metricsMgr
+  })
+
+  return {
+    ...actualModule,
+    SegmentMetricsManager: MockedClass,
+  }
+})
+
+function generateNewSession(changes = {}): NewSession {
+  return new NewSession({
+    name: "scriptName",
+    config: {
+      gatherUsageStats: false,
+      maxCachedMessageAge: 0,
+      mapboxToken: "mapboxToken",
+      allowRunOnSave: false,
+      hideSidebarNav: false,
+      hideTopBar: false,
+    },
+    customTheme: {
+      primaryColor: "red",
+      fontFaces: [],
+    },
+    initialize: {
+      userInfo: {
+        installationId: "installationId",
+        installationIdV3: "installationIdV3",
+      },
+      environmentInfo: {
+        streamlitVersion: "streamlitVersion",
+        pythonVersion: "pythonVersion",
+      },
+      sessionStatus: {
+        runOnSave: false,
+        scriptIsRunning: false,
+      },
+      sessionId: "sessionId",
+      isHello: false,
+    },
+    appPages: [
+      { pageScriptHash: "page_script_hash", pageName: "streamlit_app" },
+    ],
+    pageScriptHash: "page_script_hash",
+    mainScriptPath: "path/to/file.py",
+    scriptRunId: "script_run_id",
+    fragmentIdsThisRun: [],
+    ...changes,
+  })
+}
+
+describe("AppNavigation", () => {
+  let hostCommunicationMgr: HostCommunicationManager
+  let metricsMgr: SegmentMetricsManager
+  let onUpdatePageUrl: PageUrlUpdateCallback
+  let onPageNotFound: PageNotFoundCallback
+  let appNavigation: AppNavigation
+
+  beforeEach(() => {
+    hostCommunicationMgr = new HostCommunicationManager({
+      sendRerunBackMsg: () => {},
+      closeModal: () => {},
+      stopScript: () => {},
+      rerunScript: () => {},
+      clearCache: () => {},
+      sendAppHeartbeat: () => {},
+      setInputsDisabled: () => {},
+      themeChanged: () => {},
+      pageChanged: () => {},
+      isOwnerChanged: () => {},
+      jwtHeaderChanged: () => {},
+      hostMenuItemsChanged: () => {},
+      hostToolbarItemsChanged: () => {},
+      hostHideSidebarNavChanged: () => {},
+      sidebarChevronDownshiftChanged: () => {},
+      pageLinkBaseUrlChanged: () => {},
+      queryParamsChanged: () => {},
+      deployedAppMetadataChanged: () => {},
+    })
+    metricsMgr = new SegmentMetricsManager(new SessionInfo())
+    onUpdatePageUrl = jest.fn()
+    onPageNotFound = jest.fn()
+    appNavigation = new AppNavigation(
+      hostCommunicationMgr,
+      metricsMgr,
+      onUpdatePageUrl,
+      onPageNotFound
+    )
+  })
+
+  describe("MPA v1", () => {
+    it("sets appPages on new session", () => {
+      const maybeState = appNavigation.handleNewSession(generateNewSession())
+      expect(maybeState).not.toBeUndefined()
+
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
nit: this shows up enough that I think we might as well disable the eslint rule for the whole file given it's a test file anyway